src/Parser.cpp

```diff
@@ -77,29 +86,29 @@ static std::string trimBeginWithChar(const std::string& str, const char c) {
 
 std::vector<std::string> split(const std::string& str,
                                const std::string& delim) {
-    std::vector<std::string> result;
-    size_t start = 0;
-    size_t end = str.find(delim);
-    size_t delimLen = delim.length();
-    while (end != std::string::npos) {
-        std::string token = str.substr(start, end - start);
-        token = trimBeginWithChar(token, '\n');
-        if (!token.empty()) {
-            result.push_back(token);
-        }
-        start = end + delimLen;
-        end = str.find(delim, start);
+  std::vector<std::string> result;
+  size_t start = 0;
+  size_t end = str.find(delim);
+  size_t delimLen = delim.length();
+  while (end != std::string::npos) {
+    std::string token = str.substr(start, end - start);
+    token = trimBeginWithChar(token, '\n');
+    if (!token.empty()) {
+      result.push_back(token);
     }
+    start = end + delimLen;
+    end = str.find(delim, start);
+  }
   std::string token = trimBeginWithChar(str.substr(start), '\n');
   if (!token.empty()) {
     result.push_back(token);
   }
   return (result);
 }
 
```

Comment:
This function `split` can be moved to `utils.hpp` because it's shared by `Parser class` and `Server class (mode.cpp)`